src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+        private ServiceClientCredentials clientCredentials;
+        public ServiceClientCredentials ClientCredentials
+        {
+            get
+            {
+                return clientCredentials = clientCredentials ?? AzureSession.Instance.AuthenticationFactory.GetServiceClientCredentials(DefaultProfile.DefaultContext,
+                                               AzureEnvironment.Endpoint.ResourceManager);
+
+            }
+
+            set => clientCredentials = value;
+        }
+
+        /// <summary>
+        /// Graph RBAC client
+        /// </summary>
+        private IGraphRbacManagementClient graphRbacManagementClient;
+
+        public IGraphRbacManagementClient GraphRbacManagementClient
+        {
+            get
+            {
+
+                graphRbacManagementClient = graphRbacManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<GraphRbacManagementClient>(DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.Graph);
+
+                graphRbacManagementClient.TenantID = DefaultProfile.DefaultContext.Tenant.Id.ToString();
+
+                return graphRbacManagementClient;
+            }
+            set => graphRbacManagementClient = value;
+        }
+
+        /// <summary>
+        /// Authorization client
+        /// </summary>
+        private IAuthorizationManagementClient authorizationManagementClient;
+
+        public IAuthorizationManagementClient AuthorizationManagementClient
+        {
+            get
+            {
+                return authorizationManagementClient = authorizationManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<AuthorizationManagementClient>(DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+ 
+            }
+            set => authorizationManagementClient = value;
+        }
+
+        /// <summary>
+        /// ARM client
+        /// </summary>
+        private IResourceManagementClient resourceManagerClient;
+        public IResourceManagementClient ResourceManagerClient
+        { 
+            get
+            {
+                return resourceManagerClient = resourceManagerClient ?? new ResourceManagementClient(
+                                                   DefaultProfile.DefaultContext.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                                   ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+
+        #endregion Properties
+
+        #region Cmdlet Overrides
+        protected override void WriteExceptionError(Exception ex)
+        {
+            var aggEx = ex as AggregateException;
+
+            if (aggEx != null && aggEx.InnerExceptions != null)
+            {
+                foreach (var e in aggEx.Flatten().InnerExceptions)
+                {
+                    WriteExceptionError(e);
+                }
+
+                return;
+            }
+
+            throw ex;
```

Comment:
>ex [](start = 18, length = 2)

Should this write the exception as well? If not, we should probably have a comment explaining the reasons why not.

Reply:
It does write the exception but I made a change to call the base class explicitly. 

---
In reply to: [257092353](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257092353) [](ancestors = 257092353)